onnxruntime/core/optimizer/graph_transformer_factory.h

```diff
@@ -0,0 +1,110 @@
+  }
+
+  ORT_DISALLOW_COPY_ASSIGNMENT_AND_MOVE(GraphTransformerFactory);
+
+  std::vector<GraphTransformer*> GetTransformers(TransformerLevel level) const {
+    std::vector<GraphTransformer*> transformers;
+    const auto entry = level_to_string_.find(level);
+    if (entry != level_to_string_.end()) {
+      for (const auto& id : entry->second) {
+        const auto transformer = GetTransformers(id);
+        if (transformer != nullptr) {
+          transformers.push_back(transformer);
+        }
+      }
+    }
+    return transformers;
+  }
+
+  GraphTransformer* GetTransformers(std::string id) const {
+    const auto& entry = transformers_map_.find(id);
+    if (entry != transformers_map_.end()) {
+      return entry->second.get();
+    }
+
+    return nullptr;
+  }
+
+  std::vector<GraphTransformer*> GetTransformers(const std::vector<std::string>& ids) const {
+    std::vector<GraphTransformer*> transformers;
+    for (const auto& id : ids) {
+      const auto transformer = GetTransformers(id);
+      if (transformer != nullptr) {
+        transformers.push_back(transformer);
+      }
+    }
+    return transformers;
+  }
+
+  Status Register(std::unique_ptr<GraphTransformer> transformer) {
+    auto name = transformer->Name();
+    auto level = transformer->Level();
+
+    if (transformers_map_.find(name) != transformers_map_.end()) {
+      return Status(ONNXRUNTIME, FAIL, "This transformer is already registered");
+    }
+
+    transformers_map_[name] = std::move(transformer);
+    level_to_string_[level].push_back(name);
+    return Status::OK();
+  }
+
+ private:
+  void InitFactory() {    
+    Register(std::make_unique<ConvAddFusion>());
+    Register(std::make_unique<ConvMulFusion>());
+    Register(std::make_unique<ConvBNFusion>());    
+    Register(std::make_unique<UnsqueezeElimination>());
+    // TODO: ConvActivationFusion, GemmActivationFusion and MatMulAddFusion needs a fix. Enable this after bug fix
+    //Register(std::make_unique<ConvActivationFusion>());
+    //Register(std::make_unique<GemmActivationFusion>());    
+    //Register(std::make_unique<MatMulAddFusion>());
+
+    // Create node elimination transformer
+    Register(CreateRuleBasedTransformer());
+  }
+
+  std::unique_ptr<GraphTransformer> CreateRuleBasedTransformer() {
+    std::string transformer_name = "NodeEliminations";
+    auto level = TransformerLevel::Optional_L1;
+    auto graph_rewrite_rules = std::make_unique<TopDownRuleBasedTransformer>(
+        transformer_name, "Top down transformer", level,
+        std::vector<std::string>{onnxruntime::kCpuExecutionProvider});
+
+    graph_rewrite_rules->Register("Identity", std::make_unique<EliminateIdentity>());
+    graph_rewrite_rules->Register("Slice", std::make_unique<EliminateSlice>());
+
+    return graph_rewrite_rules;
+  }
+
+  std::unordered_map<std::string, std::unique_ptr<GraphTransformer>> transformers_map_;
+  std::unordered_map<TransformerLevel, std::vector<std::string>> level_to_string_;
+};
```

Comment:
nit: when you're naming something try and make the name clear about what it represents - in this case it's level to transformer names right? the 'string' in the name is meaningless which makes the code that uses it harder to understand as you have to go and find where it was populated to figure out that the strings are transformer names.